src/lib/fragments/Igv.react.js

```diff
@@ -0,0 +1,61 @@
+import React, {Component} from 'react';
+import {propTypes, defaultProps} from '../components/Igv.react';
+import igv from 'igv';
+
+/**
+ * The Igv component is an interactive genome visualization component
+ * developed by the Integrative Genomics Viewer (IGV) team. It uses an
+ * example integration of igv.js and React (https://www.npmjs.com/package/igv).
+ */
+export default class Igv extends Component {
+    componentDidMount() {
+        var igvContainer = document.getElementById(this.props.id);
+        var igvOptions = {
+            genome: this.props.genome,
+            locus: this.props.locus,
+            reference: this.props.reference,
+            minimumBases: this.props.minimumBases,
+            tracks: this.props.tracks,
+        };
+        return igv
+            .createBrowser(igvContainer, igvOptions)
+            .then(function(browser) {
+                igv.browser = browser;
+            });
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.props.genome !== prevProps.genome ||
+            this.props.minimumBases !== prevProps.minimumBases ||
+            this.props.locus !== prevProps.locus ||
+            this.props.reference !== prevProps.reference ||
+            this.props.tracks !== prevProps.tracks
+        ) {
+            igv.removeAllBrowsers();
```

Comment:
Oh this line is extra in `componentDidUpdate` - is this really what we want though? What if there are two igv components on the same page? Can we remove just this browser?

Reply:
Fixed in 8db8ccb so that only the browser in question is removed. I hadn't considered that someone might use multiple independent IGV browsers in a single app/page, but it could be a niche use, and it makes for more specific code.